pom.xml

```diff
@@ -345,6 +345,55 @@ limitations under the License.
                             <failIfNoMatch>false</failIfNoMatch>
                         </configuration>
                     </execution>
+                    <execution>
+                        <!-- sets the true.when.is.prerelease.version property to true if a prerelease version was used,
+                            empty string otherwise -->
+                        <id>build-helper-regex-is-prerelease-version2</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>regex-property</goal>
+                        </goals>
+                        <configuration>
+                            <name>true.when.is.prerelease.version</name>
+                            <value>${only.when.is.prerelease.version}</value>
+                            <regex>${project.version}</regex>
+                            <replacement></replacement>
+                            <failIfNoMatch>false</failIfNoMatch>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <!-- sets the docker.tags.majorVersion.minorVersion property to
+                            ${parsedVersion.majorVersion}.${parsedVersion.minorVersion} if a prerelease version was used,
+                            empty string otherwise -->
+                        <id>build-helper-regex-set-docker-version-tag</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>regex-property</goal>
+                        </goals>
+                        <configuration>
+                            <name>docker.tags.majorVersion.minorVersion</name>
+                            <value>${parsedVersion.majorVersion}.${parsedVersion.minorVersion}${true.when.is.prerelease.version}</value>
+                            <regex>.*true</regex>
+                            <replacement></replacement>
+                            <failIfNoMatch>false</failIfNoMatch>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <!-- sets the docker.tags.latest property to "latest" if a prerelease version was used,
+                            empty string otherwise -->
+                        <id>build-helper-regex-set-docker-latest-tag</id>
+                        <phase>validate</phase>
+                        <goals>
+                            <goal>regex-property</goal>
+                        </goals>
+                        <configuration>
+                            <name>docker.tags.latest</name>
```

Comment:
I guess we have to remove this part for `3.5-dev` and only add it for `3.6-dev` as `latest` should always point to the latest `3.6` image, but not a `3.5` one.

Reply:
This is true but to the best of my knowledge not a new issue. I haven't been able to run the old plugin in my environment so I can't verify this with 100% certainty but `latest` is the default tag for the old plugin and my understanding is that a new `latest` tag gets pushed with every deploy. I believe that currently we are reliant on the fact that typically 3.5.x gets deployed first, and is immediately followed by 3.6.x. I'm not aware of any way to resolve this automatically as there is no awareness within maven if the current deployment is for the "old branch" (3.5) and should not have a latest tag, or if it's for the "new branch" (3.6) and needs a latest tag.

I implemented it this way in order to best replicate the old behaviour. Perhaps it would be better to always have the latest tag disabled unless the release manager sets something like `-Ddocker.latest` during the deployment.

Reply:
Yes, there is no way to handle this automatically. That is why we only have the logic to tag `latest` present on the `3.6-dev` branch and on `master` right now, but not on the `3.5-dev` branch. That way, we can't accidentally tag a Docker image built on `3.5-dev` as `latest`.

Reply:
Ahh I wasn't paying close attention to the differences between branches and missed that. I will comment out the `latest` tag piece for `3.5-dev`, and it can be un-commented when it gets merged up.

Alternatively I can remove it entirely here and submit an additional PR for 3.6-dev.

Reply:
I think commenting out makes sense as it clearly documents this on all branches :-)